types.go

```diff
@@ -15,54 +15,62 @@ import (
 
 const defaultGradientAbsTol = 1e-6
 
-// EvaluationType is used by a Method to specify the objective-function
-// information needed at an x location.
-type EvaluationType uint
+// Operation represents the set of operations requested by Method at each
+// iteration. It is a bitmap of *Iteration and *Evaluation constants.
+type Operation uint64
 
-// Evaluation types can be composed together using the binary or operator, for
-// example 'FuncEvaluation | GradEvaluation' to evaluate both the function
-// value and the gradient.
+// Supported Operations. Individual operations must NOT be combined together by
+// the binary OR operator except for the various *Evaluations.
 const (
-	NoEvaluation   EvaluationType = 0
-	FuncEvaluation EvaluationType = 1 << (iota - 1)
+	// NoOperation does not specify any operation.
+	NoOperation Operation = 0
+	// InitIteration is sent to Recorder to indicate the initial location. All
```

Comment:
We don't send Operations to Methods. Previously NelderMead returned InitIteration for the evaluation of the initial simplex, but I think the new framework is cleaner and more consistent. Evaluations of the initial simplex are simply evaluations and when it has been evaluated, MajorIteration is returned because a satisfactory minimum might have been found. So there is no place for returning InitIteration either.